api/src/utils/media/xlsx/validation/xlsx-validation.ts

```diff
@@ -0,0 +1,75 @@
+import { SUBMISSION_MESSAGE_TYPE } from '../../../../constants/status';
+import { CSVWorkBook, WorkBookValidator } from '../../csv/csv-file';
+
+export type ParentChildKeyMatchValidatorConfig = {
+  submission_required_files_validator: {
+    name?: string;
+    description?: string;
+    child_worksheet_name: string;
+    parent_worksheet_name: string;
+    column_names: string[];
+  };
+};
+
+/**
+ * For a specified parent sheet, child sheet, and set of parent and child columns, adds an error on each cell in the
+ * child sheet whose key in the corresponding row belonging to the parent sheet cannot be found.
+ *
+ * Note: If the cell is empty, this check will be skipped. Use the `getRequiredFieldsValidator` validator to assert
+ * required fields.
+ *
+ * @param {KeyMatchValidatorConfig} [config]
+ * @return {*}  {CSVValidator}
+ */
+export const getParentChildKeyMatchValidator = (config?: ParentChildKeyMatchValidatorConfig): WorkBookValidator => {
+  return (csvWorkbook: CSVWorkBook) => {
+    if (!config) {
+      return csvWorkbook;
+    }
+
+    const { child_worksheet_name, parent_worksheet_name, column_names } = config.submission_required_files_validator;
+
+    const parentWorksheet = csvWorkbook.worksheets[parent_worksheet_name];
+    const childWorksheet = csvWorkbook.worksheets[child_worksheet_name];
+
+    if (!parentWorksheet || !childWorksheet) {
+      return csvWorkbook;
+    }
+
+    // Encodes the column values for a worksheet at a given row into a string, which is used for
+    // comparison with another worksheet
+    const serializer = (rowObject: object): string => {
+      return column_names.map((columnName: string) => (rowObject[columnName] as string).trim()).join('|');
+    };
+
+    const parentRowObjects = parentWorksheet.getRowObjects();
+    const childRowObjects = childWorksheet.getRowObjects();
+
+    const parentSerializedRows = parentRowObjects.map(serializer);
+
+    // Add an error for each cell containing a dangling key reference in the child worksheet
+    childRowObjects
+      .map(serializer)
+      .map((serializedRow: string, rowIndex: number) => {
+        return !serializedRow || parentSerializedRows.includes(serializedRow) ? -1 : rowIndex;
+      })
+      .filter((rowIndex: number) => rowIndex >= 1)
```

Comment:
Should this be `rowIndex >= 0` ?

Reply:
Oops, thought I was being clever by over skipping the header row - which is implicitly not included. Will change this to `rowIndex >= 0`